include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -3604,6 +3604,11 @@ struct OrtApi {
    *   "qnn_saver_path": File path to the QNN Saver backend library. If specified, QNN Saver will be enabled and will
    *   dump QNN API calls to disk for replay/debugging. QNN Saver produces incorrect model inference results and
    *   may alter model/EP partitioning. Use only for debugging.
+   *   "htp_graph_finalization_optimization_mode": Set the optimization mode for graph finalization on the HTP backend. Available options:
+   *     - "0": Default.
+   *     - "1": Faster preparation time, less optimal graph.
```

Comment:
note: These descriptions were adapted from the QNN SDK documentation.

Reply:
i wonder how much 0 (default) differs from 2 and 3.  i.e. should QNN EP use 2 or 3 as the default?

Reply:
My sense from our exchange with QC is that the effectiveness of this option is model-dependent. Because of the tradeoff between graph optimality and preparation time, it may be appropriate to use the QNN default and tweak it if necessary. For our partner's model, modes 2/3 were roughly the same.

Reply:
ok we can discuss a bit more with them.  they may suggest something more conservative than us :)
I think most cases we care more about the resulting inference latency vs the additional prepare time (which becomes moot once we serialize the context binary). but maybe there will be cases where the initial prepare time still matters.